hack/ci-utils/go_generate.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+set -o errexit
+
+
+main() {
+
+    # Variables checked for timeout values, defaults used if unset
+    soft_timeout="${GO_GENERATE_SOFT_TIMEOUT:-20m}"
+    hard_timeout="${GO_GENERATE_HARD_TIMEOUT:-21m}"
+
+    if [[ -n $CI ]]; then
+        set -x
+    fi
+
+
+    # Declare local before using command substitution so that local doesn't cover up timeout's exit code.
+    local timeout_exitcode go_generate
+    # go generate doesn't support timeout itself, so --preserve-status wouldn't help us as it would simply be the status timeout sent to go.
+    go_generate="go generate ./..."
+    echo "Running ${go_generate} sending SIGTERM after ${soft_timeout} and SIGKILL after ${hard_timeout}"
+    echo "${go_generate}"
+     # shellcheck disable=SC2086
+    timeout --foreground --kill-after="${hard_timeout}" -v -s SIGTERM "${soft_timeout}" ${go_generate}
+    timeout_exitcode=$?
+
+    # Capture expected exit codes
+    case $timeout_exitcode in
+    124)
+        echo "Command ${go_generate} timed out."
+        ;;
+
```

Comment:
```suggestion
```
Ultimate nit - there's some extra whitespace around the script I think can be removed.